app/models/concerns/host_searching.rb

```diff
@@ -109,13 +113,20 @@ def filter_by_compliance(_filter, operator, value)
     end
 
     def filter_by_compliance_score(_filter, operator, score)
-      ids = ::Host.includes(:test_result_profiles).select do |host|
-        host.compliance_score.public_send(operator, score.to_f)
+      unless NUM_OPERATORS.include?(operator)
+        raise ActiveRecord::StatementInvalid
       end
-      ids = ids.pluck(:id).map { |id| "'#{id}'" }
-      return { conditions: '1=0' } if ids.empty?
 
-      { conditions: "hosts.id IN(#{ids.join(',')})" }
+      profiles = RequestStore.store['scoped_search_context']
+
+      raise ScopedSearch::QueryNotSupported if profiles.nil?
+
+      hosts = ::TestResult.where("score #{operator} ?", score.to_f)
```

Comment:
Can it handle `NaN` values as not valid ones, meaning that they would not be selected in any search query?

Reply:
Let me try...

Reply:
So, I guess `NaN` is handled by PG as [infinity](https://www.youtube.com/watch?v=dL3AiuTsbOU) so if you're searching for a score in an interval, you will never actually get it as a result. However, if you are looking for all the results that are larger than 0, you will also have the `NaN` one :shrug:

Reply:
Exactly. A test result score with a `NaN` (or inf.) isn't a correct one, therefore I think it should not be returned if one wants score 90 and up.

Reply:
Well, that would require some messy hacks which we could do in a followup issue/PR